File: kalkulator.cpp
#include <iostream>

using namespace std;

void tampilkanMenu() {
    cout << "==== KALKULATOR C++ Tim B ====" << endl;
    cout << "1. Penjumlahan (+)" << endl;
    cout << "2. Pengurangan (-)" << endl;
    cout << "3. Perkalian (*)" << endl;
    cout << "4. Pembagian (/)" << endl;
    cout << "5. Modulus (%)" << endl;
    cout << "0. Keluar" << endl;
    cout << "=========================" << endl;
}
//Test Problem 1
void kalkulator() {
    int pilihan;
    double a, b;

    do {
        tampilkanMenu();
        cout << "Pilih operasi (0-7): ";
        cin >> pilihan;

        switch (pilihan) {
            case 1:
                cout << "Masukkan dua angka: ";
                cin >> a >> b;
                cout << "Hasil: " << a + b << endl;
                break;
            case 2:
                cout << "Masukkan dua angka: ";
                cin >> a >> b;
                cout << "Hasil: " << a - b << endl;
                break;
            case 3:
                cout << "Masukkan dua angka: ";
                cin >> a >> b;
                cout << "Hasil: " << a * b << endl;
                break;
            case 4:
                cout << "Masukkan dua angka: ";
                cin >> a >> b;
                if (b == 0)
                    cout << "Error: Pembagian dengan nol!" << endl;
                else
                    cout << "Hasil: " << a / b << endl;
                break;
            case 5:
                int x, y;
                cout << "Masukkan dua angka (bilangan bulat): ";
                cin >> x >> y;
                if (y == 0)
                    cout << "Error: Modulus dengan nol!" << endl;
                else
                    cout << "Hasil: " << x % y << endl;
                break;
            case 0:
                cout << "Keluar dari program." << endl;
                break;
            default:
                cout << "Pilihan tidak valid!" << endl;
        }

        cout << endl;
    } while (pilihan != 0);
}

int main() {
    kalkulator();
    return 0;
}

File: kalkulator.py
import math

def menu():
    print("\n=== KALKULATOR KOMPLEKS ===")
    print("1. Penjumlahan")
    print("2. Pengurangan")
    print("3. Perkalian")
    print("4. Pembagian")
    print("5. Pangkat")
    print("6. Akar Kuadrat")
    print("7. Logaritma (log basis 10)")
    print("8. Faktorial")
    print("9. Keluar")

def kalkulator():
    while True:
        menu()
        pilihan = input("Pilih operasi (1-9): ")

        if pilihan == '9':
            print("Terima kasih! Keluar dari kalkulator.")
            break

        try:
            if pilihan in ['1', '2', '3', '4', '5']:
                a = float(input("Masukkan angka pertama: "))
                b = float(input("Masukkan angka kedua: "))

                if pilihan == '1':
                    print("Hasil:", a + b)
                elif pilihan == '2':
                    print("Hasil:", a - b)
                elif pilihan == '3':
                    print("Hasil:", a * b)
                elif pilihan == '4':
                    if b == 0:
                        print("Error: Pembagian dengan nol!")
                    else:
                        print("Hasil:", a / b)
                elif pilihan == '5':
                    print("Hasil:", math.pow(a, b))

            elif pilihan == '6':
                a = float(input("Masukkan angka: "))
                if a < 0:
                    print("Error: Akar dari bilangan negatif!")
                else:
                    print("Hasil:", math.sqrt(a))

            elif pilihan == '7':
                a = float(input("Masukkan angka (>0): "))
                if a <= 0:
                    print("Error: log hanya untuk angka > 0!")
                else:
                    print("Hasil:", math.log10(a))

            elif pilihan == '8':
                a = int(input("Masukkan angka bulat >= 0: "))
                if a < 0:
                    print("Error: Faktorial hanya untuk angka >= 0!")
                else:
                    print("Hasil:", math.factorial(a))

            else:
                print("Pilihan tidak valid!")

        except ValueError:
            print("Error: Input tidak valid!")
# andra laiskodat 4665
# Jalankan program
kalkulator()
